packages/@react-aria/landmark/test/useLandmark.test.tsx

```diff
@@ -0,0 +1,802 @@
+      </Container>
+    );
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(tree.getAllByRole('textbox')[1]);
+  });
+
+  it('Should navigate to a landmark that has been added as a child to an existing landmark.', function () {
+
+    function Container({children = null}) {
+      return (
+        <div>
+          <Navigation>
+            <ul>
+              <li><a href="/home">Home</a></li>
+              <li><a href="/about">About</a></li>
+              <li><a href="/contact">Contact</a></li>
+            </ul>
+          </Navigation>
+          <Main>
+            <TextField label="Last Name" />
+            {children}
+          </Main>
+        </div>
+      );
+    }
+
+    let tree = render(<Container />);
+  
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(tree.getAllByRole('link')[0]);
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(tree.getAllByRole('textbox')[0]);
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(tree.getAllByRole('link')[0]);
+
+    tree.rerender(
+      <Container>
+        <Region>
+          <TextField label="First Name" />
+        </Region>
+      </Container>
+    );
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6', shiftKey: true});
+    fireEvent.keyUp(document.activeElement, {key: 'F6', shiftKey: true});
+    expect(document.activeElement).toBe(tree.getAllByRole('textbox')[1]);
+  });
+
+  it('Should navigate to a landmark that has been added as a parent to an existing landmark.', function () {
+
+    function Contained() {
+      return (
+        <div>
+          <Navigation>
+            <ul>
+              <li><a href="/home">Home</a></li>
+              <li><a href="/about">About</a></li>
+              <li><a href="/contact">Contact</a></li>
+            </ul>
+          </Navigation>
+          <Region>
+            <TextField label="Last Name" />
+          </Region>
+        </div>
+      );
+    }
+
+    let tree = render(<Contained />);
+  
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(tree.getAllByRole('link')[0]);
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(tree.getAllByRole('textbox')[0]);
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(tree.getAllByRole('link')[0]);
+
+    tree.rerender(
```

Comment:
just as a note, because of the way this is rendering, focus will be lost to the body, which is why we F6 into the textfield first after rerender
I think we should just drop lines 775-785? they can pretty much be counted as a different test, one that's already been covered

Reply:
This should be fixed now